### PR TITLE
Add a separate lint job in the Build and Test action

### LINKED ISSUE #635: Only run linter once on GH Actions
**Is your feature request related to a problem? Please describe.**

GitHub Actions runs our test script multiple times, for different versions of node. The test script includes linting. Lint errors are presumably going to be identical across Node versions. The result of this is that every lint error appears twice (three times in core) which contributes to notification fatigue. 

**Describe the solution you'd like**

1. Separate out the lint from the test script
2. Run linting as a separate step in the Actions pipeline, and only once
3. Lint errors should only appear once

**Describe alternatives you've considered**

We could just pick a version of node and have it skip linting for the other versions (would still require doing step 1). However, I like the idea of it being a separate action, that way you get more descriptive errors/status messages (e.g. the build succeeded but linting failed) whereas right now it is all under the same check.

**Additional context**

This isn't directly related to the release process situation we had last week, but it's in keeping with the theme of reducing false notifications so I'm adding it to the bash.

### PR DESCRIPTION
**Pull request by maribethb**
### What changed

This PR resolves https://github.com/google/blockly-samples/issues/635. 

The `build` job will now perform just build and test steps, and those steps will no longer perform linting themselves.
There is now a separate `lint` job that will just run the linter.

### Behavior before change

After opening a PR with a lint error, the lint error would show up 3-6 times (once each for build, test, and lint steps, and for each node_version). A lint error would cause the GH Actions UI to report that the "build" step failed.

### Behavior after change

Opening a PR will cause two separate jobs to run, Build and Lint. If there is a lint error but no other build problems, the "Build" action will succeed while the "Lint" action will fail. The lint error should appear only once. This provides both more information in the Actions UI and less clutter with duplicate errors.

### Notes

There is some duplication due to the fact that each job runs on a separate machine, so you have to npm install twice etc. But these jobs happen in parallel, so I'm hoping there is no noticeable change in the amount of time it takes (or perhaps it gets faster). I looked into it and there is a somewhat involved method of cacheing items between jobs, but I figured we should get the base model working before trying to go fancier.

As far as I know, there's not a way to test this before submitting it and hoping it works? I'll only submit this at a time when I can babysit and roll back if it's not working.

### CODE REVIEW

**Comment by maribethb** [comment]
So, good news: GitHub uses the version of the workflow in this PR on the PR. So the separate lint action works and we can test before submission.
Bad news: I must be passing the flag to the build/test scripts incorrectly because lint errors are still showing up from those steps. Will look at this later

Edit: "I just kept adding more dashes and it kept working!" Lots of indirection with npm -> lerna -> npm -> gulp so lots of layers of `--` made this work

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -24,11 +24,27 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+
+    - name: Npm Install Lerna
+      run: npm install -g lerna
```
Comment (samelhusseini) [inline .github/workflows/build.yml]: Why is this required? Npm install should install lerna locally and script that use it should be able to reference it.

**Review by samelhusseini — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (maribethb) [reply .github/workflows/build.yml]: I dunno, I copied this from what was already there. I assume this logic holds for both lint and build? I'll try removing it and seeing what happens but probably not today. Thanks for the quick review :)

Comment (maribethb) [reply .github/workflows/build.yml]: I took this step out of both jobs and everything still worked correctly, thanks!
